src/dashboard/src/app/client/admin/page.tsx

```diff
@@ -1,9 +1,14 @@
 'use client';
 
-import { useSecureRoute } from '@/hooks';
+import { useAuth } from '@/hooks';
+import { redirect } from 'next/navigation';
 
 export default function Page() {
-  useSecureRoute((state) => state.isOrganizationAdmin, '/');
+  const state = useAuth();
+
+  // Only allow Organization Admin role to view this page.
+  if (state.status === 'loading') return <div>Loading...</div>;
```

Comment:
I was hoping to extract this logic and put it into a shared function, but I can't seem to figure out the best way to do this because it's a conditional render.